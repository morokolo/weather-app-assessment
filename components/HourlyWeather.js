import React from "react";
import { FlatList, Image, StyleSheet, Text, View } from "react-native";
import colors from "../constants/colors";
import iconDictionary from "../utils/iconDictionary";

export default function HourlyWeather({ forecast }) {
  return (
    <View>
      <FlatList
        horizontal
        data={forecast.hourly.slice(0, 24)}
        keyExtractor={(item, index) => index.toString()}
        renderItem={(hour) => {
          const weather = hour.item.weather[0];
          var dt = new Date(hour.item.dt * 1000);
          return (
            <View style={styles.hour}>
              <Text style={styles.hourlyInfo}>
                {dt.toLocaleTimeString().replace(/:\d+ /, " ")}
              </Text>
              <Text style={styles.hourlyInfo}>
                {Math.round(hour.item.temp)}Â°C
              </Text>
              <Image
                style={styles.smallIcon}
                source={iconDictionary[weather.icon] || iconDictionary["01d"]}
              />
              <Text>{weather.description}</Text>
            </View>
          );
        }}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  hourlyInfo: {
    color: colors.white,
    fontWeight: "bold",
  },
  subtitle: {
    fontSize: 16,
    marginLeft: 15,
    fontWeight: "bold",
    color: colors.white,
  },
  hour: {
    padding: 6,
    alignItems: "center",
    backgroundColor: "#57575da6",
  },
  smallIcon: {
    width: 30,
    height: 30,
    marginBottom: 35,
  },
});
