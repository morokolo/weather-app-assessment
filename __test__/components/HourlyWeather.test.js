import "react-native";
import React from "react";
import HourlyWeather from "../../components/HourlyWeather";
import renderer from "react-test-renderer";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("should render the HourlyWeather component with   props", () => {
  let forecast = {
    current: {
      dt: 1614065857,
      sunrise: 1614005421,
      sunset: 1614045368,
      temp: 12.58,
      feels_like: 10.8,
      pressure: 1022,
      humidity: 71,
      dew_point: 7.47,
      uvi: 0,
      clouds: 20,
      visibility: 10000,
      wind_speed: 1.7,
      wind_deg: 12,
      weather: [
        {
          id: 801,
          main: "Clouds",
          description: "few clouds",
          icon: "02n",
        },
      ],
    },
    hourly: [
      {
        dt: 1614063600,
        temp: 12.58,
        weather: [
          {
            id: 801,
            main: "Clouds",
            description: "few clouds",
            icon: "02n",
          },
        ],
      },
    ],
    daily: [
      {
        dt: 1614024000,
        temp: {
          day: 14.99,
          min: 10.05,
          max: 16.96,
          night: 12.58,
          eve: 14.07,
          morn: 10.58,
        },

        weather: [
          {
            id: 800,
            main: "Clear",
            description: "clear sky",
            icon: "01d",
          },
        ],
      },
    ],
  };

  const tree = renderer.create(<HourlyWeather forecast={forecast} />).toJSON();
  expect(tree).toMatchSnapshot();
});
