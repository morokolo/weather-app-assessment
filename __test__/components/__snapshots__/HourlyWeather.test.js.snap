// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the HourlyWeather component with   props 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "dt": 1614063600,
          "temp": 12.58,
          "weather": Array [
            Object {
              "description": "few clouds",
              "icon": "02n",
              "id": 801,
              "main": "Clouds",
            },
          ],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#57575da6",
              "padding": 6,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "bold",
              }
            }
          >
            09:00:00
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "bold",
              }
            }
          >
            13
            Â°C
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "marginBottom": 35,
                "width": 30,
              }
            }
          />
          <Text>
            few clouds
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
